cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/preview/DefaultPreviewStore.java

```diff
@@ -41,22 +46,50 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.Nullable;
 
 /**
  * Default implementation of the {@link PreviewStore} that stores data in a level db table.
+ * This class also provides implementation of disk based {@link PreviewJobQueue}.
  */
-public class DefaultPreviewStore implements PreviewStore {
+public class DefaultPreviewStore implements PreviewStore, PreviewJobQueue {
```

Comment:
I think it's better not to implement the PreviewJobQueue here. The store only need method to persist on add, a remove on poll, and a list method to load all entries.

The reason is that, I would imagine there would be an in-memory queue that is backed by this store storage, in which the in memory queue get rebuilt upon startup. This makes things easier for mocking and testing, and also easier to add new functions to the queue that doesn't need to the hit the persistent storage.